### ISSUE TITLE
Cannot start Hedgemodmanager

### ISSUE THREAD

**Issue by leofleeo** [sonic2kk/steamtinkerlaunch#699]
## System Information
- SteamTinkerLaunch version: steamtinkerlaunch-v12.0
- Distribution: OpenSUSE tumbleweed
- Installation Method: Manual

## Issue Description
I cannot start hedgemodmanager

## Logs
Terminal output:
```
$ steamtinkerlaunch hmm start
/usr/bin/AppImageLauncher: /lib64/libcurl.so.4: no version information available (required by /usr/bin/../lib/x86_64-linux-gnu/appimagelauncher/libappimageupdate.so)
Gtk-Message: 23:40:53.191: Failed to load module "canberra-gtk-module"
Gtk-Message: 23:40:53.217: Failed to load module "colorreload-gtk-module"
Gtk-Message: 23:40:53.217: Failed to load module "window-decorations-gtk-module"
Checking latest available HedgeModManager Release version
Latest available version is '7.8-2' - Checking to see if we are up-to-date
HedgeModManager update is available (9c82456 -> 7.8-2) - Updating
Downloading HedgeModManager 7.8-2
Successfully downloaded HedgeModManager 7.8-2 to 'HedgeModManager.exe'
Finished installing HedgeModManager
Adding HedgeModManager to application menu and setting up link handlers
Setting up installed HedgeModManager compatible games
Running configuration for 'Sonic Frontiers'
Sonic Generations: Installing 'd3dcompiler_47' for Direct3D 11 mod support
Finished configuring installed HedgeModManager games
Starting HedgeModManager
```
[steamtinkerlaunch.log](https://github.com/sonic2kk/steamtinkerlaunch/files/10299441/steamtinkerlaunch.log)

**Comment by sonic2kk** [comment]
Hi, thanks for opening an issue!

In future, *please* provide more information. Provide as much detail as you can, as "HedgeModManager won't start" is not even nearly enough. You need to provide technical details of what you tried.

I haven't checked the log yet (Christmas day) but did you ensure you have an up-to-date Winetricks? Many distros do not package up-to-date Winetricks, you need `20220411-next` for dotnet48 to install. This is noted on the wiki along with various other troubleshooting steps, which you should try before opening an issue.

**Comment by sonic2kk** [comment]
I checked the log and it mentions that you're using the most up-to-date Winetricks, but as you did not provide information about what steps you tried I can't help you any further.

I would recommend investigating if dotnet48 is failing to install (the installDotNet logs are a good place to start, you may need to reinstall hmm to see them fyi if you did not look initially), if it's a multi-display issue, and other steps on the wiki. It may also be worth trying GE-Proton7-42, as 43 has issues with some programs such as MO2 and Vortex.

I'm not inclined to believe this is an STL issue, but a Winetricks issue or other issue that there are various steps listed to troubleshoot. There were various other issues opened in the past about HMM support, and the issue template did mention you should search through other issues and the wiki before opening your issue.

There have been far, far too many issues coming in for HMM with far too little information to troubleshoot, that all end up being repeats of the same external causes (ISP issues and dotnet48 failing). Steam**Tinker**Launch is for **tinkerers**, at least _some_ investigation on the users' part is expected.

If the dotnet installation logs suggest that dotnet48 can't install, e.g. errors about `ngen.exe` or an error status of 67, or if there are other generic errors with Winetricks, it's very likely that dotnet48 is failing. This is unrelated to SteamTinkerLaunch, and you should use another version of Proton to install dotnet48 such as Proton 5.0. I would try to install dotnet48 manually with GE-Proton7-43 and see if it fails. As mentioned on the wiki STL attempts to install dotnet48 for HMM as it's one of the required Winetricks. This appears to fail for some users for a variety of reasons unrelated to STL and has had extensive troubleshooting in other issues.

STL is not magic, and I am not a mind-reader, so this is guesswork on my part from other issues. Just stating that a program will not start is not very helpful, especially when there are steps documented to troubleshoot and other issues that have been opened in the past.

Please make sure you troubleshoot dotnet48 installation on your own before opening issues, because I (and any other projects that rely on installing dotnet48 via Winetricks) cannot resolve these failures as I am not a Wine, Winetricks or Proton developer.

tl;dr - Not enough information, but probably a dotnet48 problem unrelated to STL that there are steps to troubleshoot (use Proton 5.0 install HMM). In future browse closed issues and try to troubleshoot some more before opening issues, or if you did try to troubleshoot on your own, list the steps that you tried.

**Comment by lolcatplays** [comment]
I had the same issue. Delete the frontiers prefix and then open the game using Proton 5.0-10. Then leave it launching for a bit and cancel. Uninstall HMM and then in steamtinkerlaunch settings>category>hmm set it to run with 5.0-10. Then install steamtinkerlaunch from github. Extract and run the steamtinkerlaunch file using ./steamtinkerlaunch hmm start "nightly" (run console in it's folder).

Run the games with GE 42 not 43.

**Comment by leofleeo** [comment]
> I had the same issue. Delete the frontiers prefix and then open the game using Proton 5.0-10. Then leave it launching for a bit and cancel. Uninstall HMM and then in steamtinkerlaunch settings>category>hmm set it to run with 5.0-10. Then install steamtinkerlaunch from github. Extract and run the steamtinkerlaunch file using ./steamtinkerlaunch hmm start "nightly" (run console in it's folder).
> 
> Run the games with GE 42 not 43.

How do I delete the frontiers prefix? Also how do I uninstall HMM

**Comment by leofleeo** [comment]
I'm trying to get it up and running with generations not frontiers btw

**Comment by leofleeo** [comment]
It's detecting that I have frontiers installed

**Comment by leofleeo** [comment]
I figured out how to uninstall HMM

**Comment by sonic2kk** [comment]
The SteamTinkerLaunch help menu tells you how to uninstall HMM - `steamtinkerlaunch hmm uninstall`

@lolcatplays Thank you, your advice is correct. I haven't seen dotnet48 fail for HMM before though but as you also probably know Frontiers and HMM share the DotNet48 dependency.

Closing this issue as this is documented on the wiki, which should've been referred to first, and is related to dotnet48 and Winetricks and not STL. The HMM Linux issue itself also mentions this. It might seem harsh but opening an issue like this, which has been answered before and has been documented, is a tad frustrating.

Primarily, see #690, which was closed very recently and is what mentions the Proton 5.0 workaround. I'm not sure how often this is needed or how consistently it works, but there is other information online that suggests it works. There isn't a good way afaik for STL to default to this or guarantee Proton 5.0 will always be available, and as well as this GE-Proton7-4X seems to generally work (though 43 seems to have a few problems with some programs including MO2).
